packages/devtools_app/lib/src/standalone_ui/api/vs_code_api.dart

```diff
@@ -139,12 +139,22 @@ abstract interface class VsCodeDebugSession {
   /// - WebTest     (webdev test)
   String? get debuggerType;
 
+  /// The full path to the root of this project.
+  ///
+  /// This path might not always be available, for example:
+  ///
+  /// - When the version of Dart-Code is from before this field was added
+  /// - When a debug session was an attach and we didn't know the source
+  /// - When the program being run is a lose file without any pubspec
+  String? get projectRootPath;
```

Comment:
just to be clear, this is the path that should contain the `.dart_tool/` folder?

Reply:
It's the folder where we found the nearest `pubspec.yaml` while walking upwards. I _think_ that's always the same thing?

I've updated the comment to mention this explicitly ("The full path to the root of this project (the folder that contains the pubspec.yaml).")